database/connection_pool.md

```diff
@@ -0,0 +1,79 @@
+# Connection pool
+
+When a connection is stablished this usually means opening a TCP connection or Socket. The socket will handle one statement at a time. If a program needs to perform many queries simultaneously, or if handle concurrent requests that aim to use a database, it will need more than one active connection.
+
+Since databases are separate services from the application using them, the connections might go down, the services might be restarted, and other sort of things the program might not want to care.
+
+To address this issues usually a connection pool is a neat solution.
+
+When a database is opened with `crystal-db` there is already a connection pool working. `DB.open` returns a `DB::Database` object which manage the whole connection pool and not just a connection.
+
+```crystal
+DB.open("mysql://root@localhost/test") do |db|
+  # db is a DB::Database
+end
+```
+
+When executing statements using `db.query`, `db.exec`, `db.scalar`, etc. what happens is:
```

Comment:
Se siente rara la lista de pasos (que se leen como en 2da persona) con esta intro. 

Yo pondría: When executing [..] the algorithm goes: